### ISSUE TITLE
Allow dynamic S3 credentials

### ISSUE THREAD

**Issue by patrick-riemer** [CulturalMe/meteor-slingshot#147]
I would like to use several S3 buckets with several IAM users dynamically. But now the key and secret has to be defined in the Meteor settings which makes this impossible. The credentials are stored on the server.

Any chance this might be added in the near future? Is this possible at all when the S3 credentials are stored on the server and selected dynamically?

**Comment by gsuess** [comment]
> But now the key and secret has to be defined in the Meteor settings which makes this impossible.

That is not true. You can define credentials by setting the `AWSAccessKeyId` and `AWSSecretAccessKey`.

See https://github.com/CulturalMe/meteor-slingshot#aws-s3-slingshots3storage

**Comment by shantanubhadoria** [comment]
+1 to this feature
@gsuess You are right but even if we set it in the directive, its not dynamic. The region and bucket keys should accept a function in lieu of a String so that different users files may be stored in different regions/buckets closest to their location.

```
region: function () {
  // Return region value based on this.userId
},

bucket: function () {
  // Return bucket value based on this.userId
},
```

**Comment by raza2022** [comment]
@shantanubhadoria have you achieve this with any workaround,  i need the same functionality.
